src/ui/windows/TogglDesktop/TogglDesktop/ui/views/Timeline.xaml

```diff
@@ -205,6 +217,7 @@
                         <ItemsControl.ItemTemplate>
                             <DataTemplate>
                                 <togglDesktop:TimelineTimeEntryBlock Panel.ZIndex="1" DataContext="{Binding Path=Value}"
+                                                                     MouseDown="OnTimeEntryBlockMouseDown"

```

Comment:
MouseDown - this looks hacky. All the single-clicking, dragging, and resizing can be potentially captured.
ContextMenuOpening seems a better option here.

It'd also be great to make it more modular, e.g. by extracting it to Behavior with an attached property and OneWayToSource binding to ActiveTimeEntryBlock, not sure if it's possible here.